Core/Barcelona/Queries/IMDPersistence.swift

```diff
@@ -249,30 +273,30 @@ public func BLLoadChatItems(withGraph graph: [String: ([String], IMServiceStyle)
     if graph.count == 0 {
         return []
     }
-    
+
     let guids = graph.values.flatMap(\.0)
     let (buffer, remaining) = IMDPersistenceMarshal.partialBuffer(guids)
-    
+
     guard let guids = remaining else {
         return await buffer.value
     }
-    
+
     let refs = BLCreateIMItemFromIMDMessageRecordRefs(BLLoadIMDMessageRecordRefsWithGUIDs(guids))
     let items = refs.dictionary(keyedBy: \.id)
 
     let values = graph.mapValues { (guids, service) in
         (guids.compactMap { items[$0] }, service)
     }
 
-    let pendingIngestion = Future<[ChatItem], Never> { resolve in
-        Task {
-            let results: [ChatItem] = await values.asyncMap {
+    let pendingIngestion = Task<[ChatItem], Never> {
```

Comment:
wait, can you actually await results from Tasks? i thought that spawning a task functioned in the same way as DispatchQueue.async, as in that you can’t return a value from them.

Reply:
Yeah it's somewhat unexpected